--- rtadvd.c.orig	2015-03-02 23:05:13.000000000 +0000
+++ rtadvd.c	2015-03-03 09:33:27.000000000 +0000
@@ -1119,13 +1119,15 @@
 	 */
 	ifi = if_indextoifinfo(pi->ipi6_ifindex);
 	if (ifi->ifi_rainfo == NULL) {
-		syslog(LOG_INFO,
-		    "<%s> received RA from %s on non-advertising"
-		    " interface(%s)",
-		    __func__,
-		    inet_ntop(AF_INET6, &from->sin6_addr, ntopbuf,
-			sizeof(ntopbuf)), if_indextoname(pi->ipi6_ifindex,
-			ifnamebuf));
+		if (!(ifi->ifi_nd_flags & ND6_IFF_ACCEPT_RTADV)) {
+			syslog(LOG_INFO,
+		    	"<%s> received RA from %s on interface not set to accept RA"
+		   	 " interface(%s)",
+		   	 __func__,
+		    	inet_ntop(AF_INET6, &from->sin6_addr, ntopbuf,
+				sizeof(ntopbuf)), if_indextoname(pi->ipi6_ifindex,
+				ifnamebuf));
+		}
 		goto done;
 	}
 	rai = ifi->ifi_rainfo;
