--- sys/contrib/ipfilter/netinet/ip_nat.c.orig	2015/09/25 23:07:17	288242
+++ sys/contrib/ipfilter/netinet/ip_nat.c	2015/12/20 18:02:13	292518
@@ -5123,7 +5123,7 @@
 		ipf_fix_outcksum(0, &fin->fin_ip->ip_sum, msumd, 0);
 	}
 #if !defined(_KERNEL) || defined(MENTAT) || defined(__sgi) || \
-    defined(linux) || defined(BRIDGE_IPF)
+    defined(linux) || defined(BRIDGE_IPF) || defined(__FreeBSD__)
 	else {
 		/*
 		 * Strictly speaking, this isn't necessary on BSD
@@ -5235,7 +5235,7 @@
 		uh->uh_ulen += fin->fin_plen;
 		uh->uh_ulen = htons(uh->uh_ulen);
 #if !defined(_KERNEL) || defined(MENTAT) || defined(__sgi) || \
-    defined(linux) || defined(BRIDGE_IPF)
+    defined(linux) || defined(BRIDGE_IPF) || defined(__FreeBSD)
 		ipf_fix_outcksum(0, &ip->ip_sum, sumd, 0);
 #endif
 
